### ISSUE TITLE
Adding Firebase SPM creates IPA processing error when exporting an enterprise IPA

### ISSUE THREAD

**Issue by notethis** [firebase/firebase-ios-sdk#6726]
### Step 0: Are you in the right place?

  * For issues or feature requests related to __the code in this repository__
    file a Github issue.
    * If this is a __feature request__ please use the Feature Request template.
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the `firebase` tag.
  * For general (non-iOS) Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For backend issues, console issues, and other non-SDK help that does not fall under one
    of the above categories, reach out to
    [Firebase Support](https://firebase.google.com/support/).
  * Once you've read this section and determined that your issue is appropriate for
    this repository, please delete this section.

### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 12.2 beta
  * Firebase SDK version: 6.3.1
  * Firebase Component: Analytics (Auth, Core, Database, Firestore, Messaging, Storage, etc)
  * Component version: _____
  * Installation method: `Swift Package Manager` 

#### Steps to reproduce:

What happened? How can we make the problem occur?

I believe this is a duplicate of #6632 which I cannot re-open.

We've switched to using SPM to install FireBase Analytics for an internal app demonstrating out Design System. It is the only reason we had Cocoapods in the app, now the only only reason we have SPM in our app.  When attempting to do our master build on fastline, I noticed it failed shortly after `**ARCHIVE SUCCEEDED**` and failed on some process in IPAtool.  I attempted to create an Archive and then export it in Xcode after and got the same error.  

![Screen Shot 2020-10-13 at 7 52 35 PM](https://user-images.githubusercontent.com/1276108/95931344-856d9300-0d8e-11eb-8259-969a4ded243c.png)
 
I've attached a .zip of the logs:
[UIToolkitDemo_2020-10-13_19-52-15.241.xcdistributionlogs.zip](https://github.com/firebase/firebase-ios-sdk/files/5374937/UIToolkitDemo_2020-10-13_19-52-15.241.xcdistributionlogs.zip)

Only notable: 
```
2020-10-14 00:52:23 +0000  /Applications/Xcode-beta.app/Contents/Developer/usr/bin/ipatool exited with 1
2020-10-14 00:52:23 +0000  ipatool JSON: {
    alerts =     (
                {
            code = 3736;
            description = "Configuration issue: platform AppleTVSimulator.platform doesn't have any non-simulator SDKs; ignoring it";
            info =             {
            };
            level = WARN;
        },
                {
            code = 3736;
            description = "Configuration issue: platform WatchSimulator.platform doesn't have any non-simulator SDKs; ignoring it";
            info =             {
            };
            level = WARN;
        },
                {
            code = 3736;
            description = "Configuration issue: platform iPhoneSimulator.platform doesn't have any non-simulator SDKs; ignoring it";
            info =             {
            };
            level = WARN;
        }
    );
}
```

Alternatively when digging through the logs I'm getting when attempting to do this all in my CLI with Firebase: 

```
 quick-repo-test(master)$ xcodebuild -exportArchive -exportOptionsPlist /var/folders/gl/m94md3jd56d22c2hf284pppsyfj4kj/T/gym_config20201013-48837-1yvf9gm.plist -archivePath '/Users/mnote/Library/Developer/Xcode/Archives/2020-10-13/UIToolkitDemo 2020-10-13 12.13.20.xcarchive' -exportPath /var/folders/gl/m94md3jd56d22c2hf284pppsyfj4kj/T/gym_output20201013-48837-78lz64
2020-10-13 12:20:31.143 xcodebuild[50539:8331399] [MT] IDEDistribution: -[IDEDistributionLogging _createLoggingBundleAtPath:]: Created bundle at path '/var/folders/gl/m94md3jd56d22c2hf284pppsyfj4kj/T/UIToolkitDemo_2020-10-13_12-20-31.142.xcdistributionlogs'.
error: exportArchive: The data couldn't be read because it isn't in the correct format.
Error Domain=NSCocoaErrorDomain Code=3840 "No value." UserInfo={NSDebugDescription=No value., NSFilePath=/var/folders/gl/m94md3jd56d22c2hf284pppsyfj4kj/T/ipatool-json-filepath-~~~yLq54X}
```

And

```
2020-10-13 17:20:32 +0000  /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- sqlite3 (LoadError)
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /Applications/Xcode.app/Contents/Developer/usr/bin/ipatool:24:in `<main>'
```

#### Relevant Code:

```
// TODO(you): code here to reproduce the problem
```

**Comment by Exey** [comment]
I've got similar
`INFO [2020-10-15 18:38:00.21]: ▸ error: exportArchive: Found an unexpected Mach-O header code: 0x72613c21
INFO [2020-10-15 18:38:00.21]: ▸ Error Domain=DVTMachOErrorDomain Code=0 "Found an unexpected Mach-O header code: 0x72613c21" UserInfo={NSLocalizedDescription=Found an unexpected Mach-O header code: 0x72613c21, NSLocalizedRecoverySuggestion=}
INFO [2020-10-15 18:38:00.21]: ▸ ** EXPORT FAILED **`

**Comment by morganchen12** [comment]
@Exey this is #6472.

**Comment by Visput** [comment]
Having the same issue. Is there a workaround yet?

**Comment by samidaik00** [comment]
@Visput - I got around this issue by doing the following steps after getting the error alert:
 - Select Show Logs
- go in that folder IDEDistributionOptionThinning.... 
- open Payload folder
- Show package contents of your Payload, in there delete the Frameworks folder
- Finally repeat the Distribute process from Organizer and I get through the normal export successfully and can manual upload my IPA to Firebase distribution

**Comment by RafaelFernandezAlv** [comment]
I have the same issue when install Firebase with SPM. The answer of @samidaik00 don't work for me. I archive the project with custom CI and I can't modify any file when compile

**Comment by RafaelFernandezAlv** [comment]
More Info. I have this issue when I use a custom dependency that has a dependency with Firebase: https://github.com/SDOSLabs/SDOSFirebase

This is the log file: [IDEDistribution.standard.log](https://github.com/firebase/firebase-ios-sdk/files/5583328/IDEDistribution.standard.log)

```
Assertion failed: Expected 2 archs in otool output:
Archive : /var/folders/g8/hlkx4tvx6w155pjx_d6mfw4h0000gn/T/IDEDistributionOptionThinning.~~~onkhTu/Payload/AppSimple2.app/Frameworks/FIRAnalyticsConnector.framework/FIRAnalyticsConnector (architecture armv7)
/var/folders/g8/hlkx4tvx6w155pjx_d6mfw4h0000gn/T/IDEDistributionOptionThinning.~~~onkhTu/Payload/AppSimple2.app/Frameworks/FIRAnalyticsConnector.framework/FIRAnalyticsConnector(aligned_new_56a81e0f0f27268f85b0cbcf65613825.o) (architecture armv7):
```

**Comment by RafaelFernandezAlv** [comment]
Hello, I have found a solution here: https://github.com/firebase/firebase-ios-sdk/issues/6472#issuecomment-694449182. This is explained in the Swift Package Manager installation guide https://github.com/firebase/firebase-ios-sdk/blob/master/SwiftPackageManager.md

**Comment by ZaniLuca** [comment]
I managed to get it working with `Crashlytics` by using this `Run Script` proposed in this comment https://github.com/firebase/firebase-ios-sdk/issues/6864#issuecomment-720306144 
I Used the solution proposed by Fedos here https://github.com/firebase/firebase-ios-sdk/issues/6472#issuecomment-747375434 to solve the `IPA Processing error`

**Comment by paulb777** [comment]
@notethis Do you still see the issue with Xcode 12.5?

**Comment by stoqn4opm** [comment]
Guys for all of you who will come after me with this issue in XCode 12.4, it is fixed in XCode 12.5 beta 3, so just download the beta and proceed as usual
